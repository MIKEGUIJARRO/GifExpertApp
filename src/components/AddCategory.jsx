import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) {
      return;
    }
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <>
      <form
      aria-label='form'
        onSubmit={(event) => {
          onSubmit(event);
        }}
      >
        <input
          className="bg-gray-200 px-2 py-1 rounded-md"
          type={'text'}
          placeholder={'Buscar Gifs'}
          value={inputValue}
          onChange={(event) => {
            onInputChange(event);
          }}
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}