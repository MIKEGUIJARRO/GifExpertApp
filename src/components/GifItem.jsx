import React from 'react'
import PropTypes from 'prop-types';

export const GifItem = ({ id, title, url }) => {
    return (
        <div className='rounded-2xl bg-white shadow-lg overflow-hidden'>
            <img src={url} alt={title} className='w-full' />
            <div className='p-4'>
                <p className='font-bold'>{title}</p>
            </div>
        </div>
    )
}

GifItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}