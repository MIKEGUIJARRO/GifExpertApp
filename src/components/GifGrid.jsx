import React, { useEffect, useState } from 'react';
import { GifItem } from '../components';
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className='pb-16'>
      <h3 className="text-4xl font-bold pb-8">{category}</h3>
      {isLoading ? <h2>Is loading...</h2> : null}
      {<div className='grid grid-cols-3 gap-8'>
        {images.map(({ id, title, url }, index) => {
          return (
            <GifItem key={id} id={id} title={title} url={url} />
          );
        })}
      </div>}
    </div>
  );
};
