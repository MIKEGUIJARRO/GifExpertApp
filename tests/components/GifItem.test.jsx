import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Tests on <GifItem />', () => {
    const title = 'Saitama';
    const url = 'https://one-punch/saitama.jpg'
    test('should match the snapshot', () => {
        const { container } = render(<GifItem id={'1234'} title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('should show the URL image and the ALT indicated', () => {
        render(<GifItem id={'1234'} title={title} url={url} />)
        //screen.debug()
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('should show the title in the component', () => {
        render(<GifItem id={'1234'} title={title} url={url} />);
        const text = screen.getByText(title);
        expect(text).toBeTruthy();
    });
});