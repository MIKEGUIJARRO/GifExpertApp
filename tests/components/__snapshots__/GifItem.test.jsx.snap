// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match the snapshot 1`] = `
<div>
  <div
    class="rounded-2xl bg-white shadow-lg overflow-hidden"
  >
    <img
      alt="Saitama"
      class="w-full"
      src="https://one-punch/saitama.jpg"
    />
    <div
      class="p-4"
    >
      <p
        class="font-bold"
      >
        Saitama
      </p>
    </div>
  </div>
</div>
`;
